docs/cypress/e2e/links.cy.ts

```diff
@@ -10,34 +10,43 @@ type EvtName =
   | 'REQUESTING'
   | 'RETURNING'
   | 'NO_HREF';
-let allLinks: string[] = [];
-const numberOfLinks = 121;
-const requestedLinks: Set<string> = new Set();
 
-before(() => {
-  cy.task('readSitemapLinks').then((links: string[]) => {
-    allLinks = allLinks.concat(links);
-  });
-});
+describe(`All pages on Sitemap`, () => {
+  let allLinks = [];
+  const requestedLinks: Set<string> = new Set();
 
-describe('Local Sitemap', () => {
-  it('should have 119 links', () => {
-    expect(allLinks.length).to.eq(numberOfLinks);
+  before(() => {
+    cy.request('sitemap.xml').then((response) => {
+      allLinks = Cypress.$(response.body)
+        .find('loc')
+        .toArray()
+        .map((el) => el.innerText)
+        .map((link) =>
+          link
+            .replace(`${BASE_URL}/`, '')
+            .replace('https://www.dev.ui.docs.amplify.aws/', '')
+            .replace('https://ui.docs.amplify.aws/', '')
+        );
+    });
   });
-});
 
-describe(`All pages on Sitemap`, () => {
-  it.each(numberOfLinks)('all links on page %k should work', (i: string) => {
```

Comment:
Here's the trade-off. `allLinks` is empty when `it.each` sets up.

Reply:
Also, since it's a giant test, which takes a lot of memory, the Amplify Hosting Build pipeline failed on my personal account. Tried a few times, never succeed, but it works on my local machine.

But I wonder if we can try it in our Dev since its build's been quicker and less issue than my personal account.